src/io/parsers/SDFParser.js

```diff
@@ -0,0 +1,309 @@
+import Parser from './Parser';
+import chem from '../../chem';
+import * as THREE from 'three';
+import SDFStream from './SDFStream';
+import Assembly from '../../chem/Assembly';
+import ResidueType from '../../chem/ResidueType';
+
+const
+  Complex = chem.Complex,
+  Element = chem.Element,
+  Bond = chem.Bond,
+  Molecule = chem.Molecule;
+
+const chargeMap = [0, 3, 2, 1, 0, -1, -2, -3];
+const bondsMap = [
+  Bond.BondType.UNKNOWN, Bond.BondType.COVALENT, Bond.BondType.COVALENT, Bond.BondType.COVALENT, Bond.BondType.AROMATIC
+];
+
+const sdfAndMolRegexp = /.*(M\s\sEND).*|.*(^$$$$).*|.*>\s+<(.+)>.*/;
+const sdfRegExp = /.*($$$$).*|.*>\s+<(.+)>.*/;
+
+const fileFormat = {SDF: 'sdf', MOL: 'mol'};
+
+const possibleNameTags = ['PUBCHEM_IUPAC_TRADITIONAL_NAME', /PUBCHEM_(.+)_NAME/, /(.+)name/, /(.+)NAME/];
+const possibleIDTags = ['PUBCHEM_COMPOUND_CID', 'id', 'ID', /.*CID/, /.*ID/, /.*id/];
+const possibleTitleTags = ['msg', 'MSG', 'message', 'title', 'description', 'desc'];
+const tagsNames = ['name', 'id', 'title'];
+const tags = {name:  possibleNameTags, id: possibleIDTags, title: possibleTitleTags};
+
+export default class SDFParser extends Parser {
+  constructor(data, options) {
+    super(data, options);
+    this._format = 'sdf';
+    this._complex = null;
+    this._chain = null;
+    this._residue = null;
+    this._molecules = null;
+    this._metadata = {};
+    this._metadata.molecules = [];
+    this._currentMolProps = [];
```

Comment:
If we use it as a dictionary, we should initialize it as an object, not an array.